sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Set method name regex for the forbidden method.
+     * @param methodName regex for the method name
+     * @throws CheckstyleException when methodName is not a valid regex
+     */
+    public void setMethodName(Pattern methodName) throws CheckstyleException {
+        this.methodName = methodName;
+    }
+
+    /**
+     * Set arg count regex for the forbidden method, can be null.
+     * @param argumentCount regex for argument count
+     * @throws CheckstyleException when argumentCount is not a valid regex
+     */
+    public void setArgumentCount(Pattern argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+    }
+
+    @Override
+    public void init() {
+        if (methodName == null) {
+            throw new IllegalStateException(String.format(
+                "empty value for methodName (%s) is not allowed in %s",
+                methodName, CLASS_NAME));
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                final String methodNameInCode;
+                final int numArgsInCode;
+                if (dot.getType() == TokenTypes.IDENT) {
+                    methodNameInCode = dot.getText();
+                    numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
+                }
+                // method that looks like: obj.method()
+                else {
+                    methodNameInCode = dot.getLastChild().getText();
+                    numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                }
+                checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Check if the method/constructor call against defined rules.
+     * @param ast ast for the call
+     * @param name ruleName of the the method
+     * @param argCount number of arguments of the method
+     */
+    private void checkForbiddenMethod(DetailAST ast, String name, int argCount) {
+        final String argCountStr = Integer.toString(argCount);
+        final boolean methodNameMatches = methodName.matcher(name).matches();
+        final boolean argCountMatches = argumentCount == null || argumentCount.pattern().equals("")
```

Comment:
to avoid this extra expression, in the setter for `argumentCount` why don't you just assign `null` if they give an empty pattern.

Reply:
Done.